### DISCUSSION TITLE
How UI. Markdown Code Blocks Adapt to Pages

### DISCUSSION THREAD

**Discussion by xiaozhiagi** [zauberzeug/nicegui#1310] [Q&A] [answered]
### Question

In ui.markdown, long code blocks cannot adapt to page width and line breaks. How can I set the code block to scroll horizontally and vertically to solve the problem of code being too long

**Comment by falkoschindler** [comment]
Here is an example that simulates a container (`ui.card`) which is too narrow for the code blocks contained in a Markdown element:
```py
with ui.card().classes('w-60'):
    ui.markdown('''
        Here is some code:
        ```py
        print('This is a very long line of text')
        ```
    ''').classes('w-full')
```
Not that the card has a fixed with and the Markdown element is supposed to fill the available width.

To restrict the width of code blocks to 100% of the Markdown width and to automatically use scroll bars, you can query these elements and apply Tailwind classes like this:
```py
ui.query('.codehilite').classes('w-full overflow-auto')
```

**Reply by xiaozhiagi** [reply]
This code can generate a horizontal scrollbar, but using the buttons<- and ->cannot slide left or right. You must pull down the bottom and manually slide the scrollbar to see the blocked code

**Reply by falkoschindler** [reply, marked as answer]
```py
ui.query('.codehilite').classes('w-full overflow-auto').props('tabindex=0')
```
